protocol/src/main/kotlin/tbdex/sdk/protocol/models/Rfq.kt

```diff
@@ -105,12 +218,95 @@ class Rfq private constructor(
         protocol = protocol,
         externalId = externalId
       )
-      Validator.validateData(rfqData, "rfq")
 
-      // TODO: hash `data.payinMethod.paymentDetails` and set `private`
-      // TODO: hash `data.payoutMethod.paymentDetails` and set `private`
+      val (data, privateData) = hashPrivateData(unhashedRfqData)
+
+      Validator.validateData(data, "rfq")
+
+      return Rfq(metadata, data, privateData)
+    }
+
+    /**
+     * Takes an existing Message in the form of a json string and parses it into a Message object.
+     * Validates object structure and performs an integrity check using the message signature.
+     *
+     * @param payload The message as a json string.
+     * @param requireAllPrivateData If true, validate that all private data properties are present and run integrity
+     *                              check.
+     *                              Otherwise, only check integrity of private fields which are present.
+     *                              If false or omitted, validate only the private data properties that are
+     *                              currently present in `privateData`
+     * @return The json string parsed into a Rfq.
+     * @throws IllegalArgumentException if the payload is not valid json.
+     * @throws IllegalArgumentException if the payload does not conform to the expected json schema.
+     * @throws IllegalArgumentException if the payload signature verification fails.
+     * @throws IllegalArgumentException if Rfq.privateData does not match Rfq.data
+     */
+    fun parse(payload: String, requireAllPrivateData: Boolean = false): Rfq {
+      // TODO: Ensure that Message.parse() also validates private data
```

Comment:
is this todo to be done in this PR? 

also, i'm curious what happens if i call GET /exchanges and we do the parsing via `Message.parse()` on this new RFQ.

Reply:
see https://github.com/TBD54566975/tbdex-kt/blob/main/httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt#L322

Reply:
I was thinking of addressing this TODO after #224 lands. Looks like #224 has been approved, so I'll rebase it, merge, and resolve this TODO in this PR.